demo/src/main/scala/starwars/StarWarsData.scala

```diff
@@ -0,0 +1,182 @@
+      friends: => Option[List[Character]],
+      primaryFunction: Option[String]
+    ) = new Droid(id, name, appearsIn, primaryFunction)(friends)
+  }
+  // #model_types
+
+  // #model_values
+  lazy val LukeSkywalker: Character =
+    Human(
+      id = "1000",
+      name = Some("Luke Skywalker"),
+      friends = Some(List(HanSolo, LeiaOrgana, C3PO, R2D2)),
+      appearsIn = Some(List(Episode.NEWHOPE, Episode.EMPIRE, Episode.JEDI)),
+      homePlanet = Some("Tatooine"))
+  lazy val DarthVader: Character =
+    Human(
+      id = "1001",
+      name = Some("Darth Vader"),
+      friends = Some(List(WilhuffTarkin)),
+      appearsIn = Some(List(Episode.NEWHOPE, Episode.EMPIRE, Episode.JEDI)),
+      homePlanet = Some("Tatooine"))
+  lazy val HanSolo: Character =
+    Human(
+      id = "1002",
+      name = Some("Han Solo"),
+      friends = Some(List(LukeSkywalker, LeiaOrgana, R2D2)),
+      appearsIn = Some(List(Episode.NEWHOPE, Episode.EMPIRE, Episode.JEDI)),
+      homePlanet = None)
+  lazy val LeiaOrgana: Character =
+    Human(
+      id = "1003",
+      name = Some("Leia Organa"),
+      friends = Some(List(LukeSkywalker, HanSolo, C3PO, R2D2)),
+      appearsIn = Some(List(Episode.NEWHOPE, Episode.EMPIRE, Episode.JEDI)),
+      homePlanet = Some("Alderaan"))
+  lazy val WilhuffTarkin: Character =
+    Human(
+      id = "1004",
+      name = Some("Wilhuff Tarkin"),
+      friends = Some(List(DarthVader)),
+      appearsIn = Some(List(Episode.NEWHOPE, Episode.EMPIRE, Episode.JEDI)),
+      homePlanet = None)
+  lazy val C3PO: Character =
+    Droid(
+      id = "2000",
+      name = Some("C-3PO"),
+      friends = Some(List(LukeSkywalker, HanSolo, LeiaOrgana, R2D2)),
+      appearsIn = Some(List(Episode.NEWHOPE, Episode.EMPIRE, Episode.JEDI)),
+      primaryFunction = Some("Protocol"))
+  lazy val R2D2: Character =
+    Droid(
+      id = "2001",
+      name = Some("R2-D2"),
+      friends = Some(List(LukeSkywalker, HanSolo, LeiaOrgana)),
+      appearsIn = Some(List(Episode.NEWHOPE, Episode.EMPIRE, Episode.JEDI)),
+      primaryFunction = Some("Astromech"))
+
+  // The character database ...
+  val characters = List(
+    LukeSkywalker, DarthVader, HanSolo, LeiaOrgana, WilhuffTarkin, C3PO, R2D2
+  )
+  // #model_values
+
+  import Episode._
+
+  // Mapping from Episode to its hero Character
+  lazy val hero: Map[Value, Character] = Map(
+    NEWHOPE -> R2D2,
+    EMPIRE -> LukeSkywalker,
+    JEDI -> R2D2
+  )
+}
+
+object StarWarsQueryCompiler extends QueryCompiler(StarWarsSchema) {
+  // #elaborator
+  val selectElaborator = new SelectElaborator(Map(
+    StarWarsSchema.tpe("Query").dealias -> {
+      // The hero selector take an Episode argument and yields a single value. We use the
+      // Unique operator to pick out the target using the FieldEquals predicate.
+      case Select("hero", List(EnumBinding("episode", e)), child) =>
+        val episode = Episode.values.find(_.toString == e.name).get
```

Comment:
It would cover me in embarrassment ... I'll fix that.

Reply:
Looking at this again, if the query included a bogus episode, it would be caught by validation and we'd never get here. So the `find` will only fail if there's a programming error, or if there's a mismatch between the schema and the implementation. I'll push a change which handles that more defensively.